### PR TITLE
clean up some codes

### PR DESCRIPTION
**Pull request by hannahhoang2704**
Refactor code in LED.c and stepper.c  to avoid repetitive codes

### CODE REVIEW

File: LED/LED.c [hunk 1]
```diff
@@ -1,68 +1,60 @@
...
+//    }
+//}
+//
+//void blink_led(LED *led, bool state) {
+//    gpio_put(led->pin, state);
+//}
+//
+//void led_toggle(LED led) {
+//    gpio_put(led.pin, !check_led_state(&led));
+//}
```
Comment (RoniNuutinen) [inline LED/LED.c]: I think these could be deleted, rather than just commented out.

File: LED/LED.c [hunk 2]
```diff
@@ -1,68 +1,60 @@
...
 
 static uint16_t pwm_level = INIT_PWM_LEVEL;
```
Comment (RoniNuutinen) [inline LED/LED.c]: If you're reverting back to ```struct```, then ```pwm_level``` global variable is redundant:
save the level in the struct.

File: stepper/stepper.h [hunk 3]
```diff
@@ -1,7 +1,24 @@
...
+#define SPD_REDUC_MIN 850
+#define OPTO_OFFSET 148
+
+static const struct coil_struct {
+    int gpio;
+    uint8_t bit;
+} coils[COIL_COUNT] = {{IN1, 0b0001},
+                       {IN2, 0b0010},
+                       {IN3, 0b0100},
+                       {IN4, 0b1000}};
```
Comment (RoniNuutinen) [inline stepper/stepper.h:21]: What is ```.bit``` used for?

Comment (hannahhoang2704) [reply stepper/stepper.h:21]: Struct is now for both initialized pins and do bitwise with step masks. It was before in step function but now stay in header for both step function and init_stepper function

> What is `bit` used for?

File: LED/LED.c (hunk 2 above)
Comment (hannahhoang2704) [reply LED/LED.c]: > If you're reverting back to `struct`, then `pwm_level` global variable is redundant: save the level in the struct.

done!

File: stepper/stepper.h (hunk 3 above)
Comment (RoniNuutinen) [reply stepper/stepper.h:21]: Oh, I see. Good idea!

**Comment by RoniNuutinen** [comment]
Putting the macro's in .h is probably a good idea, since they have so much room relative to .c.

**Pull request merged**